examples/echo/main.go

```diff
@@ -0,0 +1,136 @@
+// This example shows using [github.com/pulumi/pulumi-go-provider] without any middleware.
+//
+// It defines one resource: `echo:index:Echo` with a single string value, which it repeats
+// back as an output.
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	p "github.com/pulumi/pulumi-go-provider"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/resource"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/tokens"
+)
+
+func main() {
+
+	echoType := tokens.NewTypeToken(tokens.NewModuleToken("echo", "index"), "Echo")
+
+	err := p.RunProvider("echo", "v0.1.0-dev", p.Provider{
+		GetSchema: func(context.Context, p.GetSchemaRequest) (p.GetSchemaResponse, error) {
+			// It is recommended to use [github.com/pulumi/pulumi-go-provider/middleware/schema.Wrap] for
+			// structured schema generation.
+			return p.GetSchemaResponse{Schema: `{
+  "name": "echo",
```

Comment:
Why not read schema.json here instead? Or remove schema.json, doesn't seem like it's used elsewhere?

Reply:
Because `schema.json` is generated from this string. This keeps the entire provider contained in this file.

We check in a `schema.json` alongside all examples so that potential authors can see the schema that corresponds to the provider. It isn't used anywhere.